integration/kubernetes/cleanup_bare_metal_env.sh

```diff
@@ -0,0 +1,22 @@
+#!/bin/bash
+#
+# Copyright (c) 2019 ARM Limited
+#
+# SPDX-License-Identifier: Apache-2.0
+
+set -o errexit
+set -o nounset
+set -o pipefail
+
+# The kubeadm reset process does not reset or clean up iptables rules
+# you must do it manually
+sudo iptables -F && iptables -t nat -F && iptables -t mangle -F && iptables -X
```

Comment:
This is a pretty hard 'drop all iptables rules' I think is it, rather than selectively dropping the k8s related rules. I suspect we have little choice, as maybe we cannot identify all k8s rules - so, I'm hoping we are not using any iptables rules for other infra on the system at the time ;-)

Reply:
yep, you're right. Let me do a few the whole CI run in my local machine to see if it will break anything. ;)

Reply:
:-) Also, you could look at the iptables at the end of a run and see what else, if anything, is in there. That's probably pretty easy to do on your local machine, and will show us if there is anything important still in the tables etc.

Reply:
I’d be afraid of running this on my local dev machines.

Reply:
@egernst ;-), well, it **should** only get run on a machine marked as a 'CI baremetal' machine, so should probably never trigger on your local box - but, yeah, @Pennyzct is checking that it will not cripple anything known to us on our existing bare metal CI machines. I'm sort of hoping that none of the cloud providers inject iptable rules into the machines 'in the background' or via some cloud init setup to support their network infra etc....
It *would* be nice if we can find a set of commands that just clear out the k8s specific bits...

Reply:
maybe we can run this only when `CI=true`, so we don't mess with dev machines?

Reply:
the call of this script is already predicated with `BAREMETAL` in the other file modified in this PR, but yes, we could go 'belt and braces' and add an `&& CI` to that as well... given the potential harm this might do if run accidentally (like, I'm pretty sure it would take out @egernst proxy setup ;-) ), let's be robust here..

Reply:
can we do iptables-save and restore instead?

Reply:
Hi~ @grahamwhaley In my local CI machine, 'drop all iptables' will affect docker indeed. 
It will also remove docker-related iptable rules. 
But all we need is to restart docker service: `systemctl restart docker`
@krsna1729 thanks for proposal. we could iptables-save in init.sh and iptables-restore after total clean. 
Wdyt????

Reply:
ack, I think save and restore is a great idea. Now, where to save.... I guess in the `init.sh` in the kubernetes dir ?

Reply:
+1 for the save and restore. Saving them as tmp file should be enough, right? we can delete it after we do the restore.